index.js

```diff
@@ -154,36 +154,37 @@ function findTag(gitPath, sha) {
 
 var LAST_TAG_CACHE = {};
 
-function findLastTagCached(gitPath, sha, commitsSinceLastTag) {
+function findLastTagCached(gitPath, sha) {
   if(!LAST_TAG_CACHE[gitPath]) {
     LAST_TAG_CACHE[gitPath] = {};
   }
 
   if(!LAST_TAG_CACHE[gitPath][sha]) {
-    LAST_TAG_CACHE[gitPath][sha] = findLastTag(gitPath, sha, commitsSinceLastTag);
+    LAST_TAG_CACHE[gitPath][sha] = findLastTag(gitPath, sha);
   }
 
   return LAST_TAG_CACHE[gitPath][sha];
 }
 
-function findLastTag(gitPath, sha, commitsSinceLastTag) {
-  commitsSinceLastTag = commitsSinceLastTag || 0;
-  var tag = findTag(gitPath, sha);
-  if(!tag) {
+function findLastTag(gitPath, sha) {
```

Comment:
Could we add a test for this?  Since we're essentially testing the order of `findTag` calls I'd consider doing something like accepting a `findTag`

```
function findLastTag(gitPath, sha, __findTag) {
  let _findTag = __findTag || findTag;
  // ...
}
```

Then in tests pass in a `findTag` that tracks the order of SHA and asserts it's searched correctly.

Reply:
Perhaps it's worth to consider using [rewire](https://www.npmjs.com/package/rewire) to monkey patch the function. It'd prefer not to change the code just for tests.